sopel/modules/currency.py

```diff
@@ -176,20 +196,32 @@ def update_rates(bot):
 
     # If we have data that is less than 24h old, return
     if time.time() - rates_updated < 24 * 60 * 60:
+        LOGGER.debug('Skipping rate update; cache is less than 24h old')
         return
 
     # Update crypto rates
+    LOGGER.debug('Updating crypto rates from %s', CRYPTO_URL)
     response = requests.get(CRYPTO_URL)
     response.raise_for_status()
     rates_crypto = response.json()
 
     # Update fiat rates
     if bot.config.currency.fixer_io_key is not None:
-        response = requests.get(FIXER_URL.format(bot.config.currency.fixer_io_key))
+        LOGGER.debug('Updating fiat rates from Fixer.io')
+        if bot.config.currency.fiat_provider != 'fixer.io':
+            # Warning about future behavior change
+            LOGGER.warning(
+                'fiat_provider is set to %s, but Fixer.io API key is present; '
+                'using Fixer anyway. In Sopel 8, fiat_provider will take precedence.',
+                bot.config.currency.fiat_provider)
+
+        response = requests.get(FIAT_PROVIDERS['fixer.io'].format(bot.config.currency.fixer_io_key))
```

Comment:
I hope FixerIO keys are always url-safe, otherwise this might fail.

Reply:
I just did exactly the same thing as what was there before.
It used to be `response = requests.get(FIXER_URL.format(bot.config.currency.fixer_io_key))`, same potential problem. Nobody has complained yet 🤷‍♂️😅

Reply:
Ha! Fair enough.